go/dpagg/count_test.go

```diff
@@ -131,13 +132,43 @@ func TestCountSerialization(t *testing.T) {
 		if !cmp.Equal(cUnchanged, cUnmarshalled, cmp.Comparer(compareCount)) {
 			t.Errorf("decode(encode(_)): when %s got %v, want %v", tc.desc, cUnmarshalled, c)
 		}
-		// Check that the original Count has its resultReturned set to true after serialization.
-		if !c.resultReturned {
-			t.Errorf("Count %v should have its resultReturned set to true after being serialized", c)
+		if c.state != Serialized {
+			t.Errorf("Count should have its state set to Serialized, got %v, want Serialized", c.state)
 		}
 	}
 }
 
+// Tests that serialization cannot be called after Result().
+func TestCountSerializationAfterResult(t *testing.T) {
+	c := getNoiselessCount()
+	c.Result()
+
+	expectedErrMessage := "Count object cannot be serialized. Reason: Noised result was already computed and returned."
```

Comment:
Yeah I see, the reason I did this was since we return an error message with reasoning now I thought it is a good idea to check that in every state an error message is being returned correctly. Moreover, this checks that the state transition are correct also. I agree I will remove it and just check the transition after because it is package private in Go but in Java I would just check that it throws an exception. Thanks!